cmd/collector/app/builder_flags_test.go

```diff
@@ -58,7 +59,18 @@ func TestCollectorOptionsWithFlags_CheckMaxReceiveMessageLength(t *testing.T) {
 	command.ParseFlags([]string{
 		"--collector.grpc-server.max-message-size=8388608",
 	})
-	c.InitFromViper(v)
+	c.InitFromViper(v, zap.NewNop())
 
 	assert.Equal(t, 8388608, c.CollectorGRPCMaxReceiveMessageLength)
 }
+
+func TestCollectorOptionsWithFlags_UseOnTagKeyConflictAsInvalid(t *testing.T) {
+	c := &CollectorOptions{}
+	v, command := config.Viperize(AddFlags)
+	command.ParseFlags([]string{
+		"--collector.use-on-tag-key-conflict=invalid",
+	})
+	c.InitFromViper(v, zap.NewNop())
+
+	assert.Equal(t, "both", c.CollectorUseOnTagKeyConflict)
+}
```

Comment:
The test is failing due to my `logger.Fatal` suggestion. This should resolve the failing test.
```suggestion
func TestCollectorOptionsWithFlags_UseOnTagKeyConflictAsInvalid(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			assert.Contains(t, r, "Invalid configuration. Allowed values: both, collector, span")
		}
	}()
	c := &CollectorOptions{}
	v, command := config.Viperize(AddFlags)
	command.ParseFlags([]string{
		"--collector.use-on-tag-key-conflict=invalid",
	})
	c.InitFromViper(v, zap.NewNop().WithOptions(zap.OnFatal(zapcore.WriteThenPanic)))
}
```